### PR TITLE
Provide error information directly when class definition loading fails

### LINKED ISSUE #559: Insufficient debug info when `Warning: The following definitions had errors while loading.`
I've been trying to update an older project to work with newer codebase. During the install, I got the following warning/error:
```
The following definitions had errors while loading.  Press Proceed to try to load them again (they may work on a second pass):
[list of classes with errors]
```
That is to be expected, but the issue is that this warning/error does not (as far as I can tell) provide any information about the individual errors in the given definitions, which is very unhelpful for someone trying to fix those errors.

There is a variable called `errorDefinitions` which contains the definitions mentioned in the message above, but no trace of the exact errors.

### PR DESCRIPTION
**Pull request by dolezvo1**
Fixes https://github.com/Metacello/metacello/issues/559

### CODE REVIEW

**Comment by dolezvo1** [comment]
I think the original code might be wrong in a slight way, as indicated by the note that a second pass might fix the errors. Perhaps this would be fixed if the terminating condition in `loadClassDefinitions` checked equivalence of the collected errors as opposed to only checking the number of the collected errors?

**Comment by Ducasse** [comment]
Please notice that I'm not sure that Pharo use the Metacello repository

**Comment by jecisc** [comment]
I changed this code in P12. I don't have time to review now but I'll review it next week

**Comment by Ducasse** [comment]
Tx Cyril :)

**Comment by jecisc** [comment]
> I think the original code might be wrong in a slight way, as indicated by the note that a second pass might fix the errors. Perhaps this would be fixed if the terminating condition in `loadClassDefinitions` checked equivalence of the collected errors as opposed to only checking the number of the collected errors?

Checking the size should be enough because we will not load definitions in the n+1 pass that was not already present in the n pass of the loading. So we don't need to check for equivalences

File: src/Monticello/MCPackageLoader.class.st [hunk 1]
```diff
@@ -242,16 +241,19 @@ MCPackageLoader >> warnAboutDependencies [
...
 				 cr.
-			 errorDefinitions do: [ :ea |
+			 errorsDict keysDo: [ :ea |
 				 s
 					 space;
 					 space;
 					 nextPutAll: ea summary;
+					 nextPutAll: ' (';
+					 nextPutAll: ((errorsDict at: ea) description asString);
+					 nextPutAll: ')';
```
Comment (jecisc) [inline src/Monticello/MCPackageLoader.class.st:257]:
```suggestion
			 errorsDict keysAndValuesDo: [ :definition :error |
				 s
					 space;
					 space;
					 nextPutAll: definition summary;
					 nextPutAll: ' (';
					 nextPutAll: error description asString;
					 nextPutAll: ')';
```

**Review by jecisc — APPROVED** [review]
I just have one little code cleaning I suggested, except that, the changes seems good to me

File: src/Monticello/MCPackageLoader.class.st (hunk 1 above)
Comment (Ducasse) [reply src/Monticello/MCPackageLoader.class.st:257]: @jecisc may be we should do your changes and integrate this PR

Comment (jecisc) [reply src/Monticello/MCPackageLoader.class.st:257]: I don't have the rights to push my changes to this PR. I'll just integrate it. My changes were for readability only

**Pull request merged**